chart/performance.tex

```diff
@@ -57,23 +57,23 @@
     area legend,
     nodes near coords={},
   ] coordinates {
-    (0, 1)
-    (1, 1)
-    (2, 3)
-    (3, 5)
-    (4, 10)
+    (0, 1.3)
+    (1, 1.2)
+    (2, 3.2)
+    (3, 4.4)
+    (4, 9.1)
   };
   \addplot[
     black,
     fill=std,
     area legend,
     nodes near coords=\pgfmathsetmacro{\input}{{\entries}[\coordindex]}\input,
   ] coordinates {
-    (0, 16)
-    (1, 14)
-    (2, 20)
-    (3, 18)
-    (4, 24)
+    (0, 22)
+    (1, 22)
+    (2, 29)
+    (3, 27)
+    (4, 34)
```

Comment:
Looking more into this significant regression in libstd times, it seems to be relatively recent but spread over multiple Rust releases. Here are the recent u64::MAX timings (the others follow a similar ratio):

- Rust 1.50&ndash;1.82: 23ns
- Rust 1.83&ndash;1.86: 26ns
- Rust 1.87&ndash;1.90-beta: 28ns
- Rust 1.91-nightly: 34ns

The regression in 1.91 bisects to nightly-2025-08-29.

Reply:
Based on `RUSTFLAGS=-Ctarget-cpu=native cargo bisect-rustc --start 2025-08-28 --end 2025-08-29 --prompt -- bench bench_std_fmt::bench_u64_max` it's https://github.com/rust-lang/rust/pull/145877.